src/zenml/cli/stack_components.py

```diff
@@ -1290,34 +1390,69 @@ def destroy_stack_component_command(
                 f"'{name_id_or_prefix}'.",
             )
 
-        # if the component's labels don't have a key created_by
-        # equal to 'recipe', then destroy cannot be called on it
-        if (
-            not component.labels
-            or "created_by" not in component.labels
-            or component.labels["created_by"] != "recipe"
-        ):
+        # Check if the component was created by a recipe
+        if not component.component_spec_path:
             cli_utils.error(
                 f"Cannot destroy stack component {component.name}. It "
                 "was not created by a recipe.",
             )
 
-        try:
-            GitStackRecipesHandler().get_stack_recipes(
-                f"{component.labels['cloud']}-modular"
-            )[0]
-        except KeyError:
-            cli_utils.error(
-                "The backing recipe for this component is missing. "
-                "Please run `zenml stack recipe pull` to fix this."
+        cli_utils.verify_mlstacks_prerequisites_installation()
+        from mlstacks.constants import MLSTACKS_PACKAGE_NAME
+
+        # spec_files_dir: str = component.component_spec_path
+        component_spec_path: str = component.component_spec_path
+        stack_name: str = os.path.basename(
+            os.path.dirname(component_spec_path)
+        )
+        stack_spec_path: str = (
+            f"{os.path.dirname(component_spec_path)}/stack-{stack_name}.yaml"
+        )
+        tf_definitions_path: str = f"{click.get_app_dir(MLSTACKS_PACKAGE_NAME)}/terraform/{provider}-modular"
+
+        cli_utils.declare(
+            "Checking Terraform definitions and spec files are present..."
+        )
+        verify_spec_and_tf_files_exist(stack_spec_path, tf_definitions_path)
+
+        from mlstacks.utils import terraform_utils
+
+        cli_utils.declare(
+            f"Destroying component '{component.name}' using Terraform..."
+        )
+        terraform_utils.destroy_stack(
+            stack_path=stack_spec_path, debug_mode=debug_mode
+        )
+        cli_utils.declare(
+            f"Component '{component.name}' successfully destroyed."
+        )
+
+        if cli_utils.confirmation(
+            f"Would you like to delete the associated ZenML "
+            f"component '{component.name}'?\nThis will delete the stack "
+            "component registered with ZenML."
+        ):
```

Comment:
Same consideration as above - why we might want to keep orphaned resources in ZenML?